bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/BaseThingHandler.java

```diff
@@ -673,4 +675,20 @@ protected void changeThingType(ThingTypeUID thingTypeUID, Configuration configur
         }
     }
 
+    /**
+     * Create a {@link ChannelBuilder} which is preconfigured with values from the given channel type.
+     *
+     * @param channelId the ID of the channel to be created
+     * @param channelTypeUID the channel type UID for which the channel should be created
+     * @return a preconfigured ChannelBuilder
+     * @throw {@link IllegalArgumentException} if the referenced channel type is not known
+     */
+    protected ChannelBuilder createChannelBuilder(String channelId, ChannelTypeUID channelTypeUID) {
```

Comment:
Do you really think that we should introduce this method on the BaseThingHandler?
Or wouldn't it suffice to provide this through the callback? After all, this method has nothing really to do with the current handler instance, but it is rather a helper/convenience/utility method that is provided by the framework.

Reply:
```java
public abstract class BaseThingHandler implements ThingHandler {
    // ...
    private @Nullable ThingHandlerCallback callback;
    // ...
}
```
How to you suggest using it in subclasses?

Reply:
The same way as subclasses are supposed to use the callback for other things. ("Not at all" does not seem to be a valid answer...?)

Reply:
Sorry, somehow I don't get it. Do you suggest to expose the callback instead to subclasses?